lib/types.nix

```diff
@@ -27,6 +27,14 @@ let
           adding [$e] to the end of the key.
         '';
       };
+      persistent = lib.mkOption {
```

Comment:
And what about 
```nix
somekey = {
  value = "myval";
  persistent = "true";
};
```
Which value should be set on activation?

Reply:
As of the latest version I am pretty sure this isn't possible to do. It won't allow values when persistence is enabled as it doesn't really make sense.

Reply:
that's great to hear :)

question:
doesn't that mean there are only 4 options left(?):
1. no value -> persistent by default
2. immutable and a value
3. immutable without a value (should that be possible?)
4. persistent and no value, explicitly

if option 3 should not be possible we could do something like: (just a suggestion, maybe for later)

```nix
somekey = null; # means persistent, is the default
someotherkey = "value"; # makes it immutable
```

then we do not need the immutable/persistent/value flags anymore (?)

Reply:
3 shouldn't have been possible, but there wasn't any checks in the code which made it illegal (but just before you commented I added a check for this as well :)). There also is shell-expansion of variables, which pretty much is the same as immutability. I also think it is possible to the explicitly setting the key to `null` which should remove the key from the file altogether. There's plenty of options and combinations, but I think we have gotten rid of most of the combinations that don't make sense.